internal/server/config.go

```diff
@@ -533,6 +534,7 @@ func (s Server) loadConfig(config Config) error {
 		// inject internal infra provider
 		config.Providers = append(config.Providers, Provider{
 			Name: models.InternalInfraProviderName,
+			Kind: models.InfraKind.String(),
```

Comment:
this actually makes any query for or excluding the infra provider much easier since the name no longer matters. instead it can filter on the kind. also should mean removing the static provider name; don't need both name and kind. might be worth a follow up

Reply:
Created #2304. We will need names for other identity providers though to cover the case of 2 identity providers of the same kind (ex: two Oktas).